### PR TITLE
Analyzer - Generate baseline given results from multiple nodes.

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Input is jsonline file with results from multiple nodes, generate baseline file according to customized rules.

### CODE REVIEW

File: superbench/analyzer/generate_baseline.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (cp5555) [inline superbench/analyzer/generate_baseline.py:1]: Would you please add a document for this function?

File: superbench/analyzer/generate_baseline.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        '--diagnosis_rule_file', type=str, default=None, required=False, help='The input path of diagnosis rule file.'
+    )
+    parser.add_argument(
+        '--summary_rule_file', type=str, default=None, required=False, help='The input path of summary rule file.'
+    )
+    args = parser.parse_args()
+    folder = args.input_dir
+    if args.algo == 'mean':
+        # simply use mean, need result_summary rules to define how to aggregate the metrics.
+        print('Generate baseine using mean of the data.')
```
Comment (cp5555) [inline superbench/analyzer/generate_baseline.py:242]: Do we need to use logger?

File: superbench/analyzer/generate_baseline.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            baseline = json.dumps(baseline, indent=2, sort_keys=True)
+            baseline = re.sub(r': \"(\d+.?\d*)\"', r': \1', baseline)
+            with open(output_dir + '/baseline.json', mode='w') as f:
+                f.write(baseline)
+
+        except Exception as e:
+            logger.error('Analyzer: generate baseline failed, msg: {}'.format(str(e)))
+
+
+if __name__ == '__main__':
```
Comment (cp5555) [inline superbench/analyzer/generate_baseline.py:215]: Would you please remove the main()? It should be the feature for sb command

File: superbench/analyzer/generate_baseline.py (hunk 1 above)
Comment (cp5555) [inline superbench/analyzer/generate_baseline.py:1]: Would you please add test for this function

File: superbench/analyzer/generate_statistic.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                            )
+            stat_df = self.calculate_statistics(aggregated_df)
+            excel_file = os.path.join(output_dir, 'benchmark_stability_stat.xlsx')
+            self.output_excel(excel_file, stat_df, digit)
+
+        except Exception as e:
+            logger.error('Analyzer: generate statisitics failed, msg: {}'.format(str(e)))
+
+
+if __name__ == '__main__':
```
Comment (cp5555) [inline superbench/analyzer/generate_statistic.py:170]: Please remove the main as well. Shall we separate PR for these two functions? We can merge generate_baseline first.

**Comment by guoshzhao** [comment]
@cp5555 This branch is used by many teams, to make sure their migration be safe, will not submit commits anymore to this branch. Please review the new one: https://github.com/microsoft/superbenchmark/pull/575

**Pull request closed without merging**